### PR TITLE
feat: promotion text on thank you page

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What are you doing in this PR?
Passes the promotion down from state into the thank you page heading.

I'm not sure we want it to look like this, but it will get us to stage 1. I wonder if a <sup>*</sup> style footnote might work better? e.g.

---

### Thank you James for supporting us with **_£5_** every month<sup>*</sup> ❤️

<sup>*</sup> For 6 months, then £10 every month.

---

☝️ has been done, I'll work with design to make sure it's all good.

## 👀 Screenshots 👀 
Screenshot 2024-03-22 at 16 33 31
Screenshot 2024-03-22 at 16 33 45

### CODE REVIEW

File: support-frontend/assets/pages/digital-subscriber-thank-you/components/thankYouHeader.tsx [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 import Subheading from './subheading';
 
+const headerTitleText = css`
+	${titlepiece.small()};
+	font-size: 24px;
+	${from.tablet} {
+		font-size: 40px;
+	}
+`;
+
```
Comment (jamesgorrie) [inline support-frontend/assets/pages/digital-subscriber-thank-you/components/thankYouHeader.tsx:18]: This was coupled to the `supporter-plus-thank-you` heading - if we want to maintain that, we should potentially create a new component.

File: support-frontend/assets/pages/supporter-plus-thank-you/components/thankYouHeader/heading.tsx [hunk 2]
```diff
@@ -1,84 +1,157 @@
...
+import type { Promotion } from 'helpers/productPrice/promotions';
 
-const MAX_DISPLAY_NAME_LENGTH = 10;
+type MonthlyProps = {
+	amount: number;
+	isoCurrency: IsoCurrency;
+	promotion?: Promotion;
+	name?: string;
+};
+function Monthly({ isoCurrency, amount, promotion, name }: MonthlyProps) {
```
Comment (paul-daniel-dempsey) [inline support-frontend/assets/pages/supporter-plus-thank-you/components/thankYouHeader/heading.tsx:18]: Maybe `MonthyHeader` gives a bit more detail?

**Review by paul-daniel-dempsey — APPROVED** [review]
Looks good, runs pefectly, assuming Helen ok with visual changes.

**Comment by jamesgorrie** [comment]
>  assuming Helen ok with visual changes

We'll work through it in another PR as this isn't being sued in `PROD` yet.

**Pull request merged**